packages/mock-addon/src/utils/faker.js

```diff
@@ -112,6 +125,18 @@ export class Faker {
         return null;
     };
 
+    matchQueryParams = (
+        searchParams,
+        requestSearchParams,
+        requestIgnoreQueryParams
+    ) => {
+        return (
+            this.ignoreQueryParams ||
+            requestIgnoreQueryParams ||
```

Comment:
I assume this `requestIgnoreQueryParams` is the value for each request. We should make this value win over the global variable.
Let's say we want to ignore query params globally and make an exception for a specific request; then, the `requestIgnoreQueryParams` value should win over the global value.

IMO it should be 

```
if (requestIgnoreQueryParams !== undefined) {
  return requestIgnoreQueryParams || arrayEquals(searchParams, requestSearchParams);
}
return  this.ignoreQueryParams || arrayEquals(searchParams, requestSearchParams)
```
Let me know your thoughts

Reply:
Oh yeah, I agree. Otherwise, people might be confused.

Reply:
Not relevant anymore because we only kept the global option.